app/controllers/users_controller.rb

```diff
@@ -31,23 +31,22 @@ def new
 
   def create
     @user = User.new(user_params_with_ip)
-
-    if @user.spam?
-      @user.is_spam = true
-      # since we don't have any relations at this point yet,
-      # only perform soft-deletion
-      @user.soft_delete
-      flash[:error] = "Hrm, robots marked you as spam. If this was done in error, please email support@alonetone.com and magic fairies will fix it right up."
-      redirect_to logout_path
-    elsif is_a_bot? && @user.valid?
-      flash[:ok] = "We just sent you an email to '#{CGI.escapeHTML @user.email}'.<br/><br/>Just click the link in the email, and the hard work is over! <br/> Note: check your junk/spam inbox if you don't see a new email right away.".html_safe
-      redirect_to login_url(already_joined: true)
-    elsif @user.valid? && @user.save_without_session_maintenance
-      @user.profile.update_attribute :user_agent, request.user_agent
-      @user.reset_perishable_token!
-      UserNotification.signup(@user).deliver_now
-      flash[:ok] = "We just sent you an email to '#{CGI.escapeHTML @user.email}'.<br/><br/>Just click the link in the email, and the hard work is over! <br/> Note: check your junk/spam inbox if you don't see a new email right away.".html_safe
-      redirect_to login_url(already_joined: true)
+    if @user.valid?
+      if @user.spam? && @user.save_without_session_maintenance
+        @user.is_spam = true
```

Comment:
This will need to be `update_attribute` or otherwise actually update the db explicitly.

Reply:
In case we don't have one already, it would be useful to have a spec that demonstrates a user marked as spam get's saved properly as spam and then is considered soft deleted.